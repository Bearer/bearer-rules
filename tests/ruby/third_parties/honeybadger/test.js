const {
  createNewInvoker,
  getEnvironment,
} = require("../../../helper.js")
const { ruleId, ruleFile, testBase } = getEnvironment(__dirname)

describe(ruleId, () => {
  const invoke = createNewInvoker(ruleId, ruleFile, testBase)

  
    test("honeybadger_breadcrumb", () => {
      const testCase = "honeybadger_breadcrumb.rb"

      const results = invoke(testCase)

      expect(results.Missing).toEqual([])
      expect(results.Extra).toEqual([])
    })
  

    test("honeybadger_context", () => {
      const testCase = "honeybadger_context.rb"

      const results = invoke(testCase)

      expect(results.Missing).toEqual([])
      expect(results.Extra).toEqual([])
    })
  

    test("honeybadger_methods", () => {
      const testCase = "honeybadger_methods.rb"

      const results = invoke(testCase)

      expect(results.Missing).toEqual([])
      expect(results.Extra).toEqual([])
    })
  

    test("honeybadger_notify", () => {
      const testCase = "honeybadger_notify.rb"

      const results = invoke(testCase)

      expect(results.Missing).toEqual([])
      expect(results.Extra).toEqual([])
    })
  
})