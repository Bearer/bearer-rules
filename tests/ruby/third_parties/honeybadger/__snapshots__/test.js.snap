// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ruby_third_parties_honeybadger honeybadger_breadcrumb 1`] = `
"{
  "high": [
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_honeybadger",
      "title": "Sensitive data sent to Honeybadger detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Honeybadger.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Honeybadger Docs](https://docs.honeybadger.io/lib/ruby/integration-guides/rails-exception-tracking/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_honeybadger",
      "line_number": 1,
      "full_filename": "/tmp/bearer-scan/honeybadger_breadcrumb.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 1,
        "end": 1,
        "column": {
          "start": 54,
          "end": 84
        }
      },
      "sink": {
        "start": 1,
        "end": 1,
        "column": {
          "start": 1,
          "end": 105
        },
        "content": "Honeybadger.add_breadcrumb(\\"Email Sent\\", metadata: { user_email: current_user.email, message: message })"
      },
      "parent_line_number": 1,
      "snippet": "Honeybadger.add_breadcrumb(\\"Email Sent\\", metadata: { user_email: current_user.email, message: message })",
      "fingerprint": "4ac2a71f9343f5925f464b72beb3339b_0",
      "old_fingerprint": "d2e8457988d84dbfabb7f2d011a4d9a4_0",
      "code_extract": "Honeybadger.add_breadcrumb(\\"Email Sent\\", metadata: { user_email: current_user.email, message: message })"
    }
  ]
}"
`;

exports[`ruby_third_parties_honeybadger honeybadger_context 1`] = `
"{
  "high": [
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_honeybadger",
      "title": "Sensitive data sent to Honeybadger detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Honeybadger.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Honeybadger Docs](https://docs.honeybadger.io/lib/ruby/integration-guides/rails-exception-tracking/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_honeybadger",
      "line_number": 1,
      "full_filename": "/tmp/bearer-scan/honeybadger_context.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "14124881-6b92-4fc5-8005-ea7c1c09592e",
        "name": "Firstname"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 1,
        "end": 1,
        "column": {
          "start": 11,
          "end": 34
        }
      },
      "sink": {
        "start": 3,
        "end": 5,
        "column": {
          "start": 1,
          "end": 3
        },
        "content": "Honeybadger.context({\\n  tags: tags\\n})"
      },
      "parent_line_number": 3,
      "snippet": "Honeybadger.context({\\n  tags: tags\\n})",
      "fingerprint": "4ac2a71f9343f5925f464b72beb3339b_0",
      "old_fingerprint": "ba018180f3a2ba14eb2e2c6266a5bd60_0",
      "code_extract": "Honeybadger.context({\\n  tags: tags\\n})"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_honeybadger",
      "title": "Sensitive data sent to Honeybadger detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Honeybadger.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Honeybadger Docs](https://docs.honeybadger.io/lib/ruby/integration-guides/rails-exception-tracking/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_honeybadger",
      "line_number": 8,
      "full_filename": "/tmp/bearer-scan/honeybadger_context.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 8,
        "end": 8,
        "column": {
          "start": 12,
          "end": 30
        }
      },
      "sink": {
        "start": 7,
        "end": 9,
        "column": {
          "start": 1,
          "end": 3
        },
        "content": "Honeybadger.context({\\n  my_data: current_user.email\\n})"
      },
      "parent_line_number": 7,
      "snippet": "Honeybadger.context({\\n  my_data: current_user.email\\n})",
      "fingerprint": "4ac2a71f9343f5925f464b72beb3339b_2",
      "old_fingerprint": "ba018180f3a2ba14eb2e2c6266a5bd60_2",
      "code_extract": "Honeybadger.context({\\n  my_data: current_user.email\\n})"
    }
  ]
}"
`;

exports[`ruby_third_parties_honeybadger honeybadger_methods 1`] = `
"{
  "high": [
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_honeybadger",
      "title": "Sensitive data sent to Honeybadger detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Honeybadger.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Honeybadger Docs](https://docs.honeybadger.io/lib/ruby/integration-guides/rails-exception-tracking/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_honeybadger",
      "line_number": 3,
      "full_filename": "/tmp/bearer-scan/honeybadger_methods.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 3,
        "end": 3,
        "column": {
          "start": 23,
          "end": 35
        }
      },
      "sink": {
        "start": 2,
        "end": 4,
        "column": {
          "start": 3,
          "end": 6
        },
        "content": "def to_honeybadger_context\\n    { user: { id: id, email: email } }\\n  end"
      },
      "parent_line_number": 2,
      "snippet": "def to_honeybadger_context\\n    { user: { id: id, email: email } }\\n  end",
      "fingerprint": "4ac2a71f9343f5925f464b72beb3339b_0",
      "old_fingerprint": "1fe89d3f8a80ec2f3dcf162cffec8a19_0",
      "code_extract": "  def to_honeybadger_context\\n    { user: { id: id, email: email } }\\n  end"
    }
  ]
}"
`;

exports[`ruby_third_parties_honeybadger honeybadger_notify 1`] = `
"{
  "high": [
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_honeybadger",
      "title": "Sensitive data sent to Honeybadger detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Honeybadger.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Honeybadger Docs](https://docs.honeybadger.io/lib/ruby/integration-guides/rails-exception-tracking/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_honeybadger",
      "line_number": 2,
      "full_filename": "/tmp/bearer-scan/honeybadger_notify.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "14124881-6b92-4fc5-8005-ea7c1c09592e",
        "name": "Fullname"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 2,
        "end": 2,
        "column": {
          "start": 30,
          "end": 39
        }
      },
      "sink": {
        "start": 28,
        "end": 35,
        "column": {
          "start": 1,
          "end": 2
        },
        "content": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
      },
      "parent_line_number": 28,
      "snippet": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)",
      "fingerprint": "4ac2a71f9343f5925f464b72beb3339b_0",
      "old_fingerprint": "9c4a47e63bd10035d0c78421e4f66784_0",
      "code_extract": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_honeybadger",
      "title": "Sensitive data sent to Honeybadger detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Honeybadger.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Honeybadger Docs](https://docs.honeybadger.io/lib/ruby/integration-guides/rails-exception-tracking/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_honeybadger",
      "line_number": 9,
      "full_filename": "/tmp/bearer-scan/honeybadger_notify.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "14124881-6b92-4fc5-8005-ea7c1c09592e",
        "name": "Firstname"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 9,
        "end": 9,
        "column": {
          "start": 11,
          "end": 34
        }
      },
      "sink": {
        "start": 28,
        "end": 35,
        "column": {
          "start": 1,
          "end": 2
        },
        "content": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
      },
      "parent_line_number": 28,
      "snippet": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)",
      "fingerprint": "4ac2a71f9343f5925f464b72beb3339b_1",
      "old_fingerprint": "9c4a47e63bd10035d0c78421e4f66784_1",
      "code_extract": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_honeybadger",
      "title": "Sensitive data sent to Honeybadger detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Honeybadger.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Honeybadger Docs](https://docs.honeybadger.io/lib/ruby/integration-guides/rails-exception-tracking/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_honeybadger",
      "line_number": 13,
      "full_filename": "/tmp/bearer-scan/honeybadger_notify.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "14124881-6b92-4fc5-8005-ea7c1c09592e",
        "name": "Firstname"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 13,
        "end": 13,
        "column": {
          "start": 5,
          "end": 29
        }
      },
      "sink": {
        "start": 28,
        "end": 35,
        "column": {
          "start": 1,
          "end": 2
        },
        "content": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
      },
      "parent_line_number": 28,
      "snippet": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)",
      "fingerprint": "4ac2a71f9343f5925f464b72beb3339b_3",
      "old_fingerprint": "9c4a47e63bd10035d0c78421e4f66784_3",
      "code_extract": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_honeybadger",
      "title": "Sensitive data sent to Honeybadger detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Honeybadger.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Honeybadger Docs](https://docs.honeybadger.io/lib/ruby/integration-guides/rails-exception-tracking/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_honeybadger",
      "line_number": 14,
      "full_filename": "/tmp/bearer-scan/honeybadger_notify.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "14124881-6b92-4fc5-8005-ea7c1c09592e",
        "name": "Lastname"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 14,
        "end": 14,
        "column": {
          "start": 5,
          "end": 27
        }
      },
      "sink": {
        "start": 28,
        "end": 35,
        "column": {
          "start": 1,
          "end": 2
        },
        "content": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
      },
      "parent_line_number": 28,
      "snippet": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)",
      "fingerprint": "4ac2a71f9343f5925f464b72beb3339b_4",
      "old_fingerprint": "9c4a47e63bd10035d0c78421e4f66784_4",
      "code_extract": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_honeybadger",
      "title": "Sensitive data sent to Honeybadger detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Honeybadger.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Honeybadger Docs](https://docs.honeybadger.io/lib/ruby/integration-guides/rails-exception-tracking/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_honeybadger",
      "line_number": 22,
      "full_filename": "/tmp/bearer-scan/honeybadger_notify.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 22,
        "end": 22,
        "column": {
          "start": 5,
          "end": 30
        }
      },
      "sink": {
        "start": 28,
        "end": 35,
        "column": {
          "start": 1,
          "end": 2
        },
        "content": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
      },
      "parent_line_number": 28,
      "snippet": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)",
      "fingerprint": "4ac2a71f9343f5925f464b72beb3339b_5",
      "old_fingerprint": "9c4a47e63bd10035d0c78421e4f66784_5",
      "code_extract": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_honeybadger",
      "title": "Sensitive data sent to Honeybadger detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Honeybadger.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Honeybadger Docs](https://docs.honeybadger.io/lib/ruby/integration-guides/rails-exception-tracking/)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_honeybadger",
      "line_number": 29,
      "full_filename": "/tmp/bearer-scan/honeybadger_notify.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "94007e1e-57d8-43e8-90f2-246236dc5dde",
        "name": "Gender"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 29,
        "end": 29,
        "column": {
          "start": 36,
          "end": 47
        }
      },
      "sink": {
        "start": 28,
        "end": 35,
        "column": {
          "start": 1,
          "end": 2
        },
        "content": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
      },
      "parent_line_number": 28,
      "snippet": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)",
      "fingerprint": "4ac2a71f9343f5925f464b72beb3339b_6",
      "old_fingerprint": "9c4a47e63bd10035d0c78421e4f66784_6",
      "code_extract": "Honeybadger.notify(\\n  \\"Something is wrong here for \\" + user.gender,\\n  class_name: \\"MyError\\",\\n  error_message: error_message,\\n  tags: tags,\\n  context: context,\\n  parameters: parameters,\\n)"
    }
  ]
}"
`;
